.github/workflows/pr.yml

```diff
@@ -110,7 +109,15 @@ jobs:
       name: Run Go Tests
       run: |
         python -m pip install --upgrade pip yq
+        go install github.com/onsi/ginkgo/v2/ginkgo@v2.0.0
         make test
+    -
+      name: Build Codecov report
```

Comment:
Note that the Codecov action was also disabled for PRs; previously we were running the tests as part of the PR check and generating coverage files, and then running them again as part of the Codecov action; it's easier to just upload the already-generated files.

I should break this out into a separate PR perhaps; maybe that would fix the codecov weirdness?

Reply:
> it's easier to just upload the already-generated files

Agreed, it'll speed things up. If we do this, it's probably worth removing the [PR trigger](https://github.com/devfile/devworkspace-operator/blob/main/.github/workflows/code-coverage.yml#L7) for the code coverage action (and only have it run on push to main), cause otherwise it will get run twice for PR's (I would imagine?).

> I should break this out into a separate PR perhaps; maybe that would fix the codecov weirdness?

It should fix the weirdness because then there will be a reference point of coverage % for `controller.cover.out` on the main branch. But if this causes other problems, then it's totally fine to leave it. Future PR's should not have this issue once the `controller.cover.out` is uploaded to codecov.

Reply:
> Agreed, it'll speed things up. If we do this, it's probably worth removing the [PR trigger](https://github.com/devfile/devworkspace-operator/blob/main/.github/workflows/code-coverage.yml#L7) for the code coverage action (and only have it run on push to main), cause otherwise it will get run twice for PR's (I would imagine?).

I thought I had already done this -- https://github.com/devfile/devworkspace-operator/pull/957/files#diff-49708f979e226a1e7bd7a68d71b2e91aae8114dd3e9254d9830cd3b4d62d4303L7-L8. Is anything else required?

My understanding is that codecov merges all uploaded files: https://docs.codecov.com/docs/merging-reports. I think this means that it should correctly compute a diff even if we were to e.g. split coverage reports per package.

Reply:
> I thought I had already done this -- [#957 (files)](https://github.com/devfile/devworkspace-operator/pull/957/files#diff-49708f979e226a1e7bd7a68d71b2e91aae8114dd3e9254d9830cd3b4d62d4303L7-L8). Is anything else required?

Oops I forgot about that when I wrote this - that should be all that's needed for the code coverage action.
 
> My understanding is that codecov merges all uploaded files: [docs.codecov.com/docs/merging-reports](https://docs.codecov.com/docs/merging-reports). I think this means that it should correctly compute a diff even if we were to e.g. split coverage reports per package.

Yes, that seems to be the case. My theory about the `controller.cover.out` not being present on master doesn't seem to hold up then. 

I think it's okay to just include the CI changes in this PR and if codecov continues to display incorrect coverage reports after this PR is merged, I will look into it.